sdk/python/kfp/dsl/_component_bridge.py

```diff
@@ -457,17 +458,20 @@ def _resolve_ir_placeholders_v2(
   original_arguments = arguments
   arguments = arguments.copy()
 
-  # Preserver input params for ContainerOp.inputs
+  # Preserve input params for ContainerOp.inputs
   input_params = list(
       set([
           param for param in arguments.values()
           if isinstance(param, _pipeline_param.PipelineParam)
       ]))
 
   for input_name, argument_value in arguments.items():
+    input_type = component_spec._inputs_dict[input_name].type
+    reference_type = None
```

Comment:
Very small naming nit: We can call this `argument_type`. "Argument" covers both reference arguments (task output reference or graph input reference) and constant arguments.